components/removed_from_channel_modal/removed_from_channel_modal.jsx

```diff
@@ -33,10 +36,9 @@ export default class RemovedFromChannelModal extends React.Component {
             BrowserStore.removeItem('channel-removed-state');
         }
 
-        var townSquare = ChannelStore.getByName(Constants.DEFAULT_CHANNEL);
         setTimeout(
             () => {
-                browserHistory.push(TeamStore.getCurrentTeamRelativeUrl() + '/channels/' + townSquare.name);
+                this.props.actions.goToLastViewedChannel();
```

Comment:
Another nit, this could be `setTimeout(this.props.actions.goToLastViewedChannel, 1);`